plugin/src/test/java/org/opensearch/ml/tools/ToolIntegrationWithLLMTest.java

```diff
@@ -72,16 +71,17 @@ public void stopMockLLM() {
     @After
     public void deleteModel() throws IOException {
         undeployModel(modelId);
+        checkForModelUndeployedStatus(modelId);
         deleteModel(client(), modelId, null);
     }
 
     @SneakyThrows
-    private void waitModelUndeployed(String modelId) {
```

Comment:
Seems like in [this PR](https://github.com/opensearch-project/ml-commons/pull/2510) we removed this line as this PR should fix this issue? Isn't that the case?

More back ground about this test here: https://github.com/opensearch-project/ml-commons/pull/2435#issuecomment-2164080756

Reply:
Thanks for bringing that up, Im checking the communication thread and notice that Zane (@zane-neo) was trying to fix the issue after aforementioned PR that omitted the `waitModelUndeployed` method.

In summary I see the following happened (Based on this [PR thread](https://github.com/opensearch-project/ml-commons/issues/2419))
1. ([May 9th](https://github.com/opensearch-project/ml-commons/issues/2419#issuecomment-2103655878)) Looks like @b4sjoo Authored a PR that fixes this class issue by updating the transportUndeployAction class. By using a callback to notify when the undeployement occured. 
2. ([July 4th] (https://github.com/opensearch-project/ml-commons/pull/2613)) Zane references a issue following the same problem and the PR thread opens again. In their own words
> The reason is deploy model is in an async approach and when client received the response, the model status in index is not guaranteed to deployed, this doesn't have a simple fix since a lot of files are involved in deploy flow.
> This issue is not able to reproduced easily on local, .. and closing this PR now. [src](https://github.com/opensearch-project/ml-commons/pull/2613#issuecomment-2272518180)

What i believe this means is that because we are working with an Async operation we have to include this method to periodically check that the model has been undeployed before deleting it to get over the `    {"error":{"root_cause":[{"type":"status_exception","reason":"Model cannot be deleted in deploying or deployed state. Try undeploy model first then delete"}],"type":"status_exception","reason":"Model cannot be deleted in deploying or deployed state. Try undeploy model first then delete"},"status":400}
` issue.